services/forms/user_form.go

```diff
@@ -429,3 +429,30 @@ func (f *U2FDeleteForm) Validate(req *http.Request, errs binding.Errors) binding
 	ctx := context.GetContext(req)
 	return middleware.Validate(errs, ctx.Data, f, ctx.Locale)
 }
+
+// WebauthnRegistrationForm for reserving an U2F name
+type WebauthnRegistrationForm struct {
+	Name          string `binding:"Required"`
+	AttType       string // attType
+	AuthType      string // authType
+	UserVer       string // userVerification
+	ResKey        string // residentKeyRequirement
+	TestExtension string // txAuthExtension
```

Comment:
The `TestExtension` should be `TxAuthSimpleExtension` according the context.

```
	testExtension := protocol.AuthenticationExtensions(map[string]interface{}{"txAuthSimple": txAuthExtension})
```

Reply:
This is not actually a Webauthn extension for reference. https://w3c.github.io/webauthn/#sctn-defined-extensions